addons/sourcemod/scripting/nd_building_fixes.sp

```diff
@@ -21,121 +31,92 @@ public Plugin myinfo =
 {
-	if (!IsValidEdict(entity))
-    	return Plugin_Handled;	
-	
-	CheckWallInsideRelay(entity);
-	return Plugin_Handled;
+    if (!ND_RoundStarted())
+        return Plugin_Continue;
+
+    if (structure == eND_RelayTower || structure == eND_WirelessRepeater)
+    {
+        // Get the team the relay tower or wireless repeater belongs to
+        int relayTeam = GetClientTeam(client);
+
+        int wallEntity = INVALID_ENT_REFERENCE;
+        while ((wallEntity = FindEntityByClassname(wallEntity, STRUCT_WALL)) != INVALID_ENT_REFERENCE)
+        {
+            // Get the team of the entity from the wall index
+            // If the wall belongs to the same team as the relay
+            int wallTeam = GetEntProp(wallEntity, Prop_Send, "m_iTeamNum");
+            if (wallTeam == relayTeam)
+            {
+                // Get the position of the wall
+                float wallPos[3];
+                GetEntPropVector(wallEntity, Prop_Data, "m_vecOrigin", wallPos);
+
+                // Compare it to the relay tower. Get the vector distance apart.
+                int distance = RoundFloat(GetVectorDistance(position, wallPos));
+                if (distance <= DISTANCE_INSIDE_RELAY)
+                {
+                    UTIL_Commander_FailureText(client, "CANNOT BUILD TOO NEAR WALL.");
+                    return Plugin_Stop;
+                }
+            }
+        }
+    }
+    else if (structure == eND_Wall)
+    {
+        // Get the team the wall belongs to
+        int wallTeam = GetClientTeam(client);
+
+        // Get the name of the relay tower (wireless repeater or relay tower)
+        char relayName[32];
+        Format(relayName, sizeof(relayName), "%s", GetRelayTowerName(wallTeam));
+
+        int relayEntity = INVALID_ENT_REFERENCE;
+        while ((relayEntity = FindEntityByClassname(relayEntity, relayName)) != INVALID_ENT_REFERENCE)
+        {
+            // Get the team of the entity from the relay index
+            // If the wall belongs to the same team as the relay
+            int relayTeam = GetEntProp(relayEntity, Prop_Send, "m_iTeamNum");
+            if (wallTeam == relayTeam)
+            {
+                // Get the position of the relay tower or wireless repeater
+                float relayPos[3];
+                GetEntPropVector(relayEntity, Prop_Data, "m_vecOrigin", relayPos);
+
+                // Compare it to the wall. Get the vector distance apart.
+                int distance = RoundFloat(GetVectorDistance(relayPos, position));
+                if (distance <= DISTANCE_INSIDE_RELAY)
+                {
+                    if (wallTeam == TEAM_CONSORT)
+                        UTIL_Commander_FailureText(client, "CANNOT BUILD TOO NEAR WIRELESS REPEATER.");
+                    else if (wallTeam == TEAM_EMPIRE)
+                        UTIL_Commander_FailureText(client, "CANNOT BUILD TOO NEAR RELAY TOWER.");
```

Comment:
This code block is not required. We can store the message in an array and index it. We already know team is valid.

Reply:
This may be what you were looking for here https://github.com/data-bomb/Redstone/commit/01d6fdc20c12dd5907d2dd86aa32cfc13eb35289 but it may be better to add a stock in nd_structures that returns the user-friendly text based on the ND_Structures enum so the same code can be used for `UTIL_Commander_FailureText` for walls as well.